src/Solvers/plan_transforms.jl

```diff
@@ -15,19 +11,177 @@ They operatore on an array with the shape of `A`, which is needed to plan
+                DiscreteTransform(forward_plan_y, Forward(), arch, grid, [2])
+            )
+
+            backward_transforms = (
+                DiscreteTransform(backward_plan_y, Backward(), arch, grid, [2]),
+                DiscreteTransform(backward_plan_x, Backward(), arch, grid, [1]),
+                DiscreteTransform(backward_plan_z, Backward(), arch, grid, [3])
+            )
+
+        elseif topo == (Bounded, Bounded, Periodic)
+            forward_plan_x = plan_forward_transform(storage, Bounded(),  [1], planner_flag)
+            forward_plan_y = plan_forward_transform(reshape(storage, (Ny, Nx, Nz)), Bounded(),  [1], planner_flag)
+            forward_plan_z = plan_forward_transform(storage, Periodic(), [3], planner_flag)
+
+            backward_plan_x = plan_backward_transform(storage, Bounded(),  [1], planner_flag)
+            backward_plan_y = plan_backward_transform(reshape(storage, (Ny, Nx, Nz)), Bounded(),  [1], planner_flag)
+            backward_plan_z = plan_backward_transform(storage, Periodic(), [3], planner_flag)
+
+            forward_transforms = (
+                DiscreteTransform(forward_plan_x, Forward(), arch, grid, [1]),
+                DiscreteTransform(forward_plan_y, Forward(), arch, grid, [2]),
+                DiscreteTransform(forward_plan_z, Forward(), arch, grid, [3])
+            )
+
+            backward_transforms = (
+                DiscreteTransform(backward_plan_z, Backward(), arch, grid, [3]),
+                DiscreteTransform(backward_plan_x, Backward(), arch, grid, [1]),
+                DiscreteTransform(backward_plan_y, Backward(), arch, grid, [2])
+            )
+
+        elseif topo == (Bounded, Bounded, Bounded)
+            forward_plan_x = plan_forward_transform(storage, Bounded(), [1], planner_flag)
+            forward_plan_y = plan_forward_transform(reshape(storage, (Ny, Nx, Nz)), Bounded(), [1], planner_flag)
+            forward_plan_z = plan_forward_transform(storage, Bounded(), [3], planner_flag)
+
+            backward_plan_x = plan_backward_transform(storage, Bounded(), [1], planner_flag)
+            backward_plan_y = plan_backward_transform(reshape(storage, (Ny, Nx, Nz)), Bounded(),  [1], planner_flag)
+            backward_plan_z = plan_backward_transform(storage, Bounded(), [3], planner_flag)
+
+            forward_transforms = (
+                DiscreteTransform(forward_plan_x, Forward(), arch, grid, [1]),
+                DiscreteTransform(forward_plan_y, Forward(), arch, grid, [2]),
+                DiscreteTransform(forward_plan_z, Forward(), arch, grid, [3])
+            )
+
+            backward_transforms = (
+                DiscreteTransform(backward_plan_x, Backward(), arch, grid, [1]),
+                DiscreteTransform(backward_plan_y, Backward(), arch, grid, [2]),
+                DiscreteTransform(backward_plan_z, Backward(), arch, grid, [3])
+            )
+        end
+
+    else
```

Comment:
Is this the case `(Periodic, Periodic, Periodic)`?

Reply:
This is the case where batching transforms is possible. It's always possible on the CPU since FFTW is awesome so it includes all topologies on the CPU.

On the GPU batching is possible when the topology is not one of `non_batched_topologies` (where an FFT is needed along dimension 2), so it includes `(Periodic, Periodic, Periodic)`, `(Periodic, Periodic, Bounded)`, and `(Bounded, Periodic, Periodic)`.

Reply:
Future generations may thank you if you put this comment in the code